.github/workflows/lint-and-typecheck.yml

```
@@ -27,6 +25,15 @@ jobs:
       - name: Install pnpm
         run: npm install -g pnpm
 
+      - name: Set up pnpm environment
+        run: |
```

actionlint (1.7.3)

shellcheck reported issue in this script: SC2155:warning:2:8: Declare and assign separately to avoid masking return values

(shellcheck)